lib/wallet/api/http/Cardano/CLI.hs

```diff
@@ -655,7 +664,56 @@ cmdWalletCreateFromMnemonic mkClient =
                 (ApiT wName)
                 (ApiT wPwd)
                 Nothing
-                Nothing
+                (Just $ ApiT rm)
+
+restorationModeOption :: Parser (RestorationMode -> a) -> Parser a
+restorationModeOption f =
+    hsubparser
+        $ fromGenesisCmd <> fromCheckpointCmd <> fromTipCmd
+  where
+    fromGenesisCmd =
+        command "from-genesis"
+            $ info (f <*> pure RestoreFromGenesis)
+            $ progDesc "Restore the wallet from genesis."
+            <> header "Restoration from genesis (full restoration)"
+            <> footer "This is the way to restore the wallet from the very beginning."
+    fromTipCmd =
+        command "from-tip"
+            $ info (f <*> pure RestoreFromTip)
+            $ progDesc "Restore the wallet from the tip."
+            <> header "Restoration from tip (no restoration)"
+            <> footer "This is the way to not restore the wallet"
+    fromCheckpointCmd =
+        command "from-checkpoint"
+            $ info (f <*> fromCheckpointP)
+            $ progDesc "Restore the wallet from a specific checkpoint."
+            <> header "Restoration from checkpoint (partial restoration)"
+            <> footerDoc (Just fromCheckpointCmdFooter)
+
+    fromCheckpointCmdFooter = vsep [
+        "This is the way to restore the wallet from a specific checkpoint."
+        , "It will create a partial view of the wallet if the wallet has \
+            \transactions before the checkpoint!"
+        , "You will need a blockheader hash and a slot number to restore from."
+        , "Example:"
```

Comment:
```    , "Example:"
        , indent 2 "cardano-wallet wallet create from-recovery-passphrase \
            \from-checkpoint \
            \--block-header-hash 95af5a1da7d43571d030c92ba82728d69c86c82007cff6ee3c8b2442ab148723 \
            \'My wallet'"
        , "Example:"
        , indent 2 "cardano-wallet wallet create from-recovery-passphrase \
            \from-checkpoint \
            \--slot-no 118332737 \
            \'My wallet'"
```
You specify EITHER block header hash OR slot number. Not both, right?

Reply:
Only both

Reply:
ok, now understand the constraint. Thanks!